pkg/proxy/winkernel/proxier.go

```diff
@@ -352,6 +353,11 @@ func (proxier *Proxier) newServiceInfo(port *v1.ServicePort, service *v1.Service
 	if err != nil {
 		preserveDIP = false
 	}
+	dsr := service.Spec.ExternalTrafficPolicy == v1.ServiceExternalTrafficPolicyTypeLocal
```

Comment:
You need to also check if the feature flag is enabled as DSR is still an alpha feature.

Reply:
@ksubrmnn Isn't that feature gate tied to the preserveDIP implementation of DSR? AFAIK this is a separate implementation based on the existing ExternalTrafficPolicy feature in the service spec. @madhanrm correct me if I'm wrong

Reply:
No it is not. That flag is for DSR, not preserve dip. PreserveDIP is not an implementation of DSR. Rather, it depends on DSR.

However, you are right in that we won't need to guard with a flag for the pre-existing ExternalTrafficPolicy feature. So this is a separate implementation.

For 1.20, you can follow up with a PR to add DSR functionality for the other loadbalancers if the feature flag is enabled.

Reply:
Please rename this variable so that it is not confused with proxier.isDSR

Reply:
> For 1.20, you can follow up with a PR to add DSR functionality for the other loadbalancers if the feature flag is enabled.

I'm not quite following on this. Could you explain what you mean here?

Also, I will change the variable to localTrafficDSR

Reply:
DSR is its own feature. PreserveDIP depends on DSR.

We have enabled DSR to work for all types of loadbalancers on the platform. However, external traffic policy only applies to loadbalancers of type Loadbalancer and of type NodePort. This PR also needs to be amended to honor that.

To add DSR to the other loadbalancer types, we would need to first check the feature flag. Then we would enable it accordingly.

Reply:
Looks like Kubernetes will block on yaml deployment if the incorrect loadbalancer type is specified with externalTrafficPolicy local so you won't have to worry about that after all.

Reply:
Resolving this as out of scope for the PR. We are continuing this conversation offline for future changes